src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -27,23 +22,23 @@ class JsonAdaptedPerson {
     private final String name;
     private final String phone;
     private final String email;
-    private final String address;
-    private final List<JsonAdaptedTag> tags = new ArrayList<>();
+    private final String classGroup;
+    private final String telegram;
+    private final String github;
 
     /**
      * Constructs a {@code JsonAdaptedPerson} with the given person details.
      */
     @JsonCreator
     public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
-            @JsonProperty("email") String email, @JsonProperty("address") String address,
-            @JsonProperty("tags") List<JsonAdaptedTag> tags) {
+            @JsonProperty("email") String email, @JsonProperty("classGroup") String classGroup,
+            @JsonProperty("telegram") String telegram, @JsonProperty("github") String github) {
         this.name = name;
         this.phone = phone;
         this.email = email;
-        this.address = address;
-        if (tags != null) {
-            this.tags.addAll(tags);
-        }
+        this.classGroup = classGroup;
+        this.telegram = telegram == null ? "" : telegram;
+        this.github = github == null ? "" : github;
```

Comment:
Would the constructor ever pass in null instead of an empty string? In that case, is it better if we use Optional instead?

Reply:
changed it to not use null and use empty string instead.

Reply:
actually some of the test cases designed to be an invalid person have missing tele and github values - so there is a possibility of null values.

I've changed this to "this.telegram = telegram"

and then I do the checks for null. the prof's code also has null checks for the other strings so I think that's how its meant to be.

Reply:
Awesome!